pkg/cli/completion.go

```diff
@@ -85,6 +86,98 @@ func (a *completion) complete(cmd *cobra.Command, args []string, toComplete stri
 	return removeExistingArgs(result, args), a.successDirective
 }
 
+// checkProjectPrefix
+// called on a completion object, which already has the completion function defined
+// this function modifies that completion function, wraps it with some logic to handle projects
+func (a *completion) checkProjectPrefix() *completion {
+	currentCompletionFunc := a.completionFunc
+	foundCompletions := []string{}
+	// Get CLI Config, necessary for building new clients on different projects
+	options := a.client.Options()
+	if options.CLIConfig == nil {
+		cfg, err := config.ReadCLIConfig()
+		options.CLIConfig = cfg
+		if err != nil {
+			fmt.Printf("could not read CLI config.\n%+v", err)
+		}
+	}
+	newCompletionFunc := func(ctx context.Context, c client.Client, toComplete string) ([]string, error) {
+		if parsedProject, after, found := strings.Cut(toComplete, "::"); found {
+			// check that the provided project exists, return a helpful error otherwise
+			foundProject, err := c.ProjectGet(ctx, parsedProject)
+			if foundProject == nil || err != nil {
+				return []string{}, nil
+			}
```

Comment:
The comment says that this is returning an error, but the `err` value being returned here is nil.

Reply:
If it doesn't find anything to complete, there isn't really an error, just no completions.